src/Features/ERDDAP/Platform/Observations/CurrentConditions/data_card.tsx

```diff
@@ -1,95 +1,153 @@
-  let filtered_datasets: PlatformDataset[] = []
+  let filteredTimeSeries: PlatformTimeSeries[] = []
 
-  data_types.forEach((data_type) => {
+  data_types.forEach((dataType) => {
     platform.properties.readings
-      .filter((reading) => data_type === reading.data_type.standard_name)
-      .forEach((reading) => filtered_datasets.push(reading))
+      .filter((reading) => dataType === reading.data_type.standard_name) // match any that are the current data type
+      .filter((reading) => (reading.time ? aDayAgo < new Date(reading.time) : false) && reading.depth < 2) // that have data in the last day, and that are near-surface
+      .forEach((reading) => filteredTimeSeries.push(reading))
   })
 
-  // Don't return anything if there are no datasets
-  if (filtered_datasets.length === 0) {
-    return null
+  // No current data or data types don't exist for platform
+  if (filteredTimeSeries.length < 1) {
+    let noFilteredTimeSeries: PlatformTimeSeries[] = []
+    data_types.forEach((dataType) => {
+      platform.properties.readings
+        .filter((reading) => dataType === reading.data_type.standard_name && reading.depth < 2)
+        .forEach((reading) => noFilteredTimeSeries.push(reading))
+    })
+
+    // Platform does not contain any of the requested data types
+    if (noFilteredTimeSeries.length < 1) {
+      return null
+    }
+
+    return <NoDataCard platform={platform} timeSeries={noFilteredTimeSeries[0]} />
   }
 
-  let reading = filtered_datasets[0]
+  const timeSeries = filteredTimeSeries[0]
 
-  let data = reading.readings
+  return <LoadDataCard timeSeries={timeSeries} platform={platform} />
+}
 
-  // If there is no current data, display a card letting us know
-  if (data.length === 0) {
-    return (
-      <Col {...cardProps} key={reading.data_type.standard_name}>
-        <Card>
-          <CardBody>No data for {reading.data_type.long_name} in the last day.</CardBody>
-        </Card>
-      </Col>
+interface LoadDataCardProps {
+  platform: PlatformFeature
+  timeSeries: PlatformTimeSeries
+}
+
+/**
+ * Load the data for a time series and connect units to the data card to display
+ * additionally filter data to the last day
+ *
+ * @param platform
+ * @param timeSeries Time series to load data for
+ */
+const LoadDataCard: React.FC<LoadDataCardProps> = ({ platform, timeSeries }) => {
+  const { isLoading, data } = useDataset(timeSeries)
+  const unit_system = useUnitSystem()
+
+  if (isLoading) {
+    return <LoadingDataCard platform={platform} timeSeries={timeSeries} />
+  }
+
+  if (data) {
+    const aDayAgo = new Date()
+    aDayAgo.setDate(aDayAgo.getDate() - 1)
+
+    const readings: ReadingTimeSeries[] = (data as DataTimeSeries).timeSeries.filter(
+      (reading) => aDayAgo < reading.time
     )
+
+    if (readings.length > 0) {
+      return (
+        <DataCardDisplay readings={readings} unit_system={unit_system} timeSeries={timeSeries} platform={platform} />
```

Comment:
Codacy found an issue: [Identifier 'unit_system' is not in camel case.](https://app.codacy.com/manual/gmri/Neracoos-1-Buoy-App/pullRequest?prid=6166252)